### ISSUE TITLE
Trainer.set_learning_rate is broken when distributed kvstore is used

### ISSUE THREAD

**Issue by eric-haibin-lin** [apache/incubator-mxnet#15919]
https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/trainer.py#L265-L275
When update_on_kvstore=True, Trainer.set_learning_rate does not modify the learning rate on the remote parameter server
